doc/index.md

```diff
@@ -46,16 +46,65 @@ file and the CREDITS file for the licensing and acknowledgements.
 --------------------------------------------
 The code is known to work on the following compilers:
 
-- clang 3.6.2
-- GCC 4.9.1
+- clang 3.6.2 and later
+- GCC 4.9.1 and later
 
 \section tutorial-quick-start Quick Start
 
 --------------------------------------------
-Range v3 is a generic library that augments the existing standard library with
-facilities for working with *ranges*. A range can be loosely thought of a pair
-of iterators, although they need not be implemented that way. Bundling begin/end
-iterators into a single object brings several benefits.
+
+## What This Library Provides
+
+Range v3 is a generic library that augments the existing standard library with facilities for working with *ranges*. This means that you do not have to fiddle with `begin` and `end` iterators directly, but pass something around, that represents both in one object. 
+
+For every STL algorithm this library provides a version that accepts a *range* instead of a pair of iterators. 
+
+All transforming and filtering algorithms are composable. This means applying an algorithm to a range yields a **lazily** evaluated representation (a so-called **view**) of the result, which can either be evaluated to obtain the results of the applied transformation or can be passed to another algorithm.  
+
+This avoids temporary results while allowing a declarative and easy-to-understand programming style. Generally this is possible without sacrificing performance.  
+
+Range v3 also provides in-place modifying algorithms e.g. for sorting. These are called **action**s.  
+
+Range v3 is a contribution to making C++ easier to write and reason about. With Range v3 one can write code this way: 
+
+~~~~~~~{.cpp}
+    std::vector<PolarCoordinate> coordinates;
+    
+    auto outsideOfReasonableRange =
+        (PolarCoordinate pc) { return pc.distance > 3.0; };
+
+    auto is_inf_or_nan =
+        [](PolarCoordinate pc) { 
+          return std::isinf(pc.distance) 
+            || std::isnan(pc.distance); 
+            };
+
+    auto filteredAndInCartesianCoordinates = coordinates
+        | remove_if(outsideOfReasonableRange)
+        | remove_if(is_inf_or_nan)
+        | transform(polarToCartesian)
+        | reverse;
+~~~~~~~
+
+
+
+
+## What is a Range?
+
+It may come as a surprise, but giving a short, comprehensive definition of a *range* is much harder than one might expect. 
+
+First of all a range is something you can iterate over (*range* used to be spelled *iterable*). One possibility to describe an iterable is given in the [`IEnumerator`  interface](https://docs.microsoft.com/de-de/dotnet/api/system.collections.ienumerator) definition of the `C#` programming language which has a `bool MoveNext()` method and a `Current` property. Due to consistency with the STL and for efficiency reasons the C++ representation differs. Everything that can be passed to `begin` and `end` can be considered to be a range. 
```

Comment:
Strike "(range used to be called iterable)". Use the term "range" consistently. The comparison to the C# `IEnumerable` interface will not help those people who don't know C#. I'd rather with stick to C++. A range is anything we can call `std::begin` and `std::end` on to retrieve an iterator and a sentinel. It can be a value or a reference, a container or a view, or even a built-in C+ array.

Reply:
> Strike "(range used to be called iterable)".

Well, I stole this information here (and found it useful to understand what it is)
![image](https://user-images.githubusercontent.com/5588692/52444088-ae547400-2b27-11e9-8422-a92a69283ceb.png)

Reply:
I agree with C# comparison to be moved into an extra section about other languages